.github/workflows/test.yml

```diff
@@ -0,0 +1,66 @@
+---
+name: "Test"
+on:
+  pull_request:
+  push:
+    branches:
+      - "main"
+jobs:
+  tox:
+    name: "Test ${{ matrix.toxenv }}"
+    runs-on: "ubuntu-18.04"
+    strategy:
+      matrix:
+        include:
+          - python-version: "3.6"
+            toxenv: "py36"
+          - python-version: "3.7"
+            toxenv: "py37"
+          - python-version: "3.8"
+            toxenv: "py38"
+          - python-version: "3.9"
+            toxenv: "py39"
+    steps:
+      - name: "Check out repository"
+        uses: "actions/checkout@v2"
+      - name: "Set up Python ${{ matrix.python-version }}"
+        uses: "actions/setup-python@v2"
+        with:
+          python-version: "${{ matrix.python-version }}"
+      - name: "Get pip cache dir"
+        id: "pip-cache"
+        run: |
+          echo "::set-output name=dir::$(pip cache dir)"
+      - name: "Cache pip packages"
+        uses: "actions/cache@v2"
+        with:
+          path: "${{ steps.pip-cache.outputs.dir }}"
+          key: "${{ runner.os }}-pip-${{ hashFiles('**/base.txt', '**/test.txt') }}"
+          restore-keys: |
+            ${{ runner.os }}-pip-
+      - name: "Install tox"
+        run: |
+          python -m pip install --upgrade pip
+          pip install tox
+      - name: "Run tox"
+        env:
+          TOXENV: ${{ matrix.toxenv }}
+        run: |
+          tox
+  lint:
+    name: "Lint"
+    runs-on: "ubuntu-18.04"
+    steps:
+      - name: "Check out repository"
+        uses: "actions/checkout@v2"
+      - name: "Set up Python"
+        uses: "actions/setup-python@v2"
+        with:
+          python-version: "3.8"
```

Comment:
I'm curious why you only target 3.8 for linting?

Reply:
It's a bit arbitrary, tbh! The idea is that the linting checks are Python version-independent but it still needs a version of Python to run.

It's also worth noting that this file, the pre-commit config, the Makefile, tox.ini file, etc. were just copy-pasted wholesale from other projects. Hence why I'd like to just make a Cookiecutter template to simplify getting all this stuff into a new repo :)